### PR TITLE
V2 guild-related CRUD operations

### PR DESCRIPTION
**Pull request by cs-balazs**
# V2 guild-related CRUD operations

The `v2Replacer` approach has been replaced, now we can just prefix an url with `/v2/`

Planned changes:
- [x] Break down role updates to smaller, more atomic operations (role update + requirement creations/updates + rolePlatform creations/updates)
- [x] Send only data that has been changed
- [x] Don't refetch guild data in onSuccess, instead use the data returned from update, create calls
- [x] rolePlatform creation now returns a `createdGuildPlatform` in case the rolePlatform creation caused a new guildPlatform to be created, this mutation should also be handled in onSuccess 
- [x] rolePlatform DELETE with new endpoint
- [x] guildPlatform DELETE with new endpoint
- [x] guild edit
- [x] role delete
- [x] guild delete
- [x] handle requirement deletions (probably just by calling the DELETE immediately, like for rolePlatforms)
- [ ] POAPs
- [ ] Remove any unnecessary guild data refetches in favour of mutating directly from the returned data

**!!!! IMPORTANT**: This branch points to dev environment, make sure to revert env changes before merging

### CODE REVIEW

File: src/components/[guild]/RoleCard/components/EditRole/hooks/useEditRole.ts [hunk 1]
```diff
@@ -1,35 +1,230 @@
...
+      updatedRequirements,
+      createdRequirements,
+      updatedRolePlatforms,
+      createdRolePlatforms,
+    }) => {
+      const deletedRequirementIds = new Set(
+        createdRequirements.flatMap((req) => req.deletedRequirements)
+      )
+
+      mutateGuild((prev) => ({
```
Comment (BrickheadJohnny) [inline src/components/[guild]/RoleCard/components/EditRole/hooks/useEditRole.ts:119]: Do we need 2 `mutateGuild`s in this file? Or could you maybe combine them?

File: src/components/create-guild/hooks/useCreateGuild.tsx [hunk 2]
```diff
@@ -20,7 +20,7 @@ const useCreateGuild = () => {
   const fetcherWithSign = useFetcherWithSign()
 
   const fetchData = async (signedValidation: SignedValdation): Promise<Guild> =>
-    fetcher("/guild", signedValidation)
+    fetcher("/v2/guilds/with-roles", signedValidation)
```
Comment (BrickheadJohnny) [inline src/components/create-guild/hooks/useCreateGuild.tsx:23]: Oh, one more note here: I think we'll need to update the `cypress/e2e/0-platformless/0-create-guild.spec.ts` test file too, it is failing rn ;)

File: src/components/create-guild/Requirements/components/RequirementEditableCard.tsx [hunk 3]
```diff
@@ -106,13 +172,14 @@ const RequirementEditableCard = ({
...
         <CloseButton
+          ref={closeButtonRef}
           position="absolute"
           top={2}
           right={2}
           color={removeButtonColor}
           borderRadius={"full"}
           size="sm"
-          onClick={() => removeRequirement(index)}
+          onClick={() => onRemove()}
```
[the remark this answers is not in the document]
Comment (cs-balazs) [reply src/components/create-guild/Requirements/components/RequirementEditableCard.tsx]: Thanks, changed in https://github.com/agoraxyz/guild.xyz/commit/10f670ad79ac4f42b03d8f469bee057eaca8f27c!

File: src/hooks/useIsV2.ts [hunk 4]
```diff
@@ -1,11 +1,24 @@
+import useUser from "components/[guild]/hooks/useUser"
+import { posthog } from "posthog-js"
 import { useFeatureFlagEnabled } from "posthog-js/react"
+import { useEffect } from "react"
 
 const V2_FEATURE_FLAG_KEY = "api-v2"
 
+// Edge config?
```
[the remark this answers is not in the document]
Comment (BrickheadJohnny) [reply src/hooks/useIsV2.ts:8]: Oh, interesting idea, I'd say we can try edge config, sure!
I checked the PostHog feature flag settings too, I thought we can set up users who should always get this flag, but unfortunately that wouldn't be as convenient as edge config.

So yeah, we could just fetch the config once on the first page load with SWRImmutable, and then just use it from SWR cache in the whole session.

**Pull request merged**